sympy/physics/continuum_mechanics/tests/test_beam.py

```diff
@@ -173,6 +173,25 @@ def test_Beam():
         b4.variable = 1
 
 
+def test_variable_second_moment():
+    E = Symbol('E')
+    I = Symbol('I')
+
+    b = Beam(4, E, Piecewise((1.5*I,x<=2), (I, x<=4)))
```

Comment:
Why can't singularity functions handle this? Not clear why Piecewise is needed at all.

Can't there be something like: `10*<x-4>**2 / E1 / I1 + 5 * <x-10>**2 / E2 / I1`?

Singularity Functions are just another way to express piecewise.

Reply:
I tried the same but if `I` is wriiten in form of  `SingularityFunction` then `integrate(self.bending_moment()/I, x)` doesn't integrate. Like for the given test:
```
>>> b.bending_moment()
80*SingularityFunction(x, 0, 0) - 20*SingularityFunction(x, 0, 1) 
+ 20*SingularityFunction(x, 4, 1)
>>> second_moment = 1.5*I1*SingularityFunction(x, 0, 0) - 1.5*I1*SingularityFunction(x, 2, 0) + I1*SingularityFunction(x, 2, 0)
>>>second_moment.subs(x,1)
1.5*I1
>>>second_moment.subs(x,3)
1.0*I1

```
but to find `slope` when we integrate their division, it doesn't return complete integral:
```
>>> integrate(b.bending_moment()/second_moment, x)
-13.3333333333333*(Integral(4*SingularityFunction(x, 0, 0)/(-1.0*SingularityFunction(x, 0, 0) + 0.333333333333333*SingularityFunction(x, 2, 0)), x) + Integral(-4*SingularityFunction(x, 4, 0)/(-1.0*SingularityFunction(x, 0, 0) + 0.333333333333333*SingularityFunction(x, 2, 0)), x) + Integral(-x*SingularityFunction(x, 0, 0)/(-1.0*SingularityFunction(x, 0, 0) + 0.333333333333333*SingularityFunction(x, 2, 0)), x) + Integral(x*SingularityFunction(x, 4, 0)/(-1.0*SingularityFunction(x, 0, 0) + 0.333333333333333*SingularityFunction(x, 2, 0)), x))/I1
```

Reply:
It was just that integration was simpler with `Piecewise`. 
So the newer way to input variable `second_moment` is as a list of tuples of the form `(Value, start, order, end)`. (similar to the way we input `load`)